data/profiles/taskrunner.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2016, EMC, Inc.
+
+"""
+Download and execute python scripts from the RackHD tasks api in a loop
+"""
+
+import urllib2
+import json
+import sys
+from time import sleep
+
+TASK_REQUEST_PERIOD = 5
+TASKS_URI = 'http://<%=server%>:<%=port%>/api/common/tasks/<%=identifier%>'
+
+json_content_type = {"Content-Type": "application/json"}
+
+def download_task_data():
+    """
+    Download a python script from the tasks API
+    """
+    _task_data = urllib2.urlopen(TASKS_URI).read()
+    return json.loads(_task_data)
+
+
+def download_script(downloadUrl):
+    """
+    Download a python script from the tasks API
+    """
+    url = urllib2.urlparse.urljoin('http://<%=server%>:<%=port%>', downloadUrl)
+    print "Downloading script at {}".format(url)
+    script = urllib2.urlopen(url).read()
+    with open('script.py', 'w') as rackhd_script:
+        rackhd_script.write(script)
+
+while True:
```

Comment:
I think we need a big `try`/`catch` block for the endless `while` loop, otherwise any exception of a particular command will cause the task runner failed and impact the following commands. I can see following cases may raise an exception:
1 - int(task_data["exit"])
2 - urllib2.Request
3 - urllib2.urlopen